src/rpdk/core/contract/resource_client.py

```diff
@@ -242,6 +278,33 @@ def assert_time(start_time, end_time, action):
             "Handler %r timed out." % action
         )
 
+    @staticmethod
+    def get_decoded_identifier_list(identifier_path):
+        list_identifier = [
+            identifier[len(identifier) - 1] for identifier in list(identifier_path)
+        ]
+        return list_identifier
+
+    @staticmethod
+    def assert_primary_identifier(primary_identifier_path, resource_model):
+        primary_identifiers_list = ResourceClient.get_decoded_identifier_list(
+            primary_identifier_path
+        )
+        for primary_identifier in primary_identifiers_list:
+            assert resource_model.get(primary_identifier) is not None
+
+    @staticmethod
+    def assert_primary_identifier_not_updated(
+        primary_identifier_path, created_model, updated_model
+    ):
+        primary_identifiers_list = ResourceClient.get_decoded_identifier_list(
+            primary_identifier_path
+        )
+        for primary_identifier in primary_identifiers_list:
```

Comment:
`assert all( condition for primary_identifier in primary_identifiers_list ) `

Reply:
changed the method to look similar to assert_primary_identifier